lib/core/error/exception.dart

```diff
@@ -0,0 +1,3 @@
+class ServerException implements Exception {}
```

Comment:
Let's follow consistent namings on files, since this file has multiple exceptions, I think it should be `exceptions.dart`